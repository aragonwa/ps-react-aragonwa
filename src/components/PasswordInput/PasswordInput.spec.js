import React from 'react';
import renderer from 'react-test-renderer';
import PasswordInput from './PasswordInput';
import {shallow} from 'enzyme';

test('toggles input type when show/hide password clicked', () => {
  const wrapper = shallow(<PasswordInput 
    htmlId="test"
    name="test"
    value=""
    onChange={()=>{}}
    showVisibilityToggle
  />);
  // Password Input should have a type of password initially
  expect(wrapper.find({type: 'password'})).toHaveLength(1);
  expect(wrapper.find({type: 'text'})).toHaveLength(0);

  wrapper.find('a').simulate('click');
  expect(wrapper.find({type: 'password'})).toHaveLength(0);
  expect(wrapper.find({type: 'text'})).toHaveLength(1);
});

test('shows password quality when enabled and password is entered',()=>{
  const tree = renderer.create(<PasswordInput 
    htmlId="test"
    name="test"
    onChange={()=>{}}
    showQuality
    value="UisIxx#3334"
  />).toJSON();
  expect(tree).toMatchSnapshot(); 
});