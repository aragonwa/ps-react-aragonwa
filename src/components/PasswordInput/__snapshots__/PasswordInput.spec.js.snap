// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows password quality when enabled and password is entered 1`] = `
<div
  style={
    Object {
      "marginBottom": 16,
    }
  }
>
  <label
    htmlFor={
      Object {
        "htmlFor": "test",
      }
    }
    style={
      Object {
        "display": "block",
      }
    }
  >
    Password
     
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
       *
    </span>
  </label>
  <input
    id="test"
    maxLength={50}
    name="test"
    onChange={[Function]}
    placeholder={undefined}
    showQuality={true}
    style={undefined}
    type="password"
    value="UisIxx#3334"
  />
</div>
`;
